### PR TITLE
Corrects falsiness assumptions

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

When converting from GRPC classes to Typescript classes, the code often uses `if (!field)` to determine if an object is undefined. However, this catches other falsey cases as well, such as if the field equals `0` or `""`. This PR corrects those assumptions where needed. 

### Context of Change

This was noticed while auditing all the Typescript wrapper classes.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

## Before / After

Falsiness assumptions have been corrected. 

## Test Plan

CI passes. 

## Future Tasks
More corrections will be made in future auditing PRs.

### CODE REVIEW

**Review by amiecorso — APPROVED** [review]
great catch on this

File: src/XRP/model/xrp-account-set.ts [hunk 1]
```diff
@@ -29,7 +29,11 @@ export default class XrpAccountSet {
     const emailHash = accountSet.getEmailHash()?.getValue_asU8()
     const messageKey = accountSet.getMessageKey()?.getValue_asU8()
     const setFlag = accountSet.getSetFlag()?.getValue()
-    if (clearFlag != undefined && setFlag && clearFlag === setFlag) {
+    if (
+      clearFlag != undefined &&
+      setFlag != undefined &&
```
Comment (tedkalaw) [inline src/XRP/model/xrp-account-set.ts:34]: `!==` please

File: src/XRP/model/xrp-currency-amount.ts [hunk 2]
```diff
@@ -19,11 +19,11 @@ export default class XrpCurrencyAmount {
     // Mutually exclusive: either drops or issuedCurrency is set in an XRPCurrencyAmount
     const issuedCurrencyAmount = currencyAmount.getIssuedCurrencyAmount()
     const xrpAmount = currencyAmount.getXrpAmount()
-    if (issuedCurrencyAmount && !xrpAmount) {
+    if (issuedCurrencyAmount != undefined && xrpAmount == undefined) {
       const issuedCurrency = XrpIssuedCurrency.from(issuedCurrencyAmount)
       return new XrpCurrencyAmount(undefined, issuedCurrency)
     }
-    if (xrpAmount && !issuedCurrencyAmount) {
+    if (xrpAmount != undefined && issuedCurrencyAmount == undefined) {
```
Comment (tedkalaw) [inline src/XRP/model/xrp-currency-amount.ts:26]: same with these `===` and `!==`

**Review by tedkalaw — CHANGES_REQUESTED** [review]
can we use the `===` operators here instead of `==`? this should technically work because undefined is only equal to itself but afaik we don't use type coercion in the sdk for these checks

**Review by tedkalaw — APPROVED** [review]

**Pull request merged**